core/src/test/scala/org/apache/spark/util/JsonProtocolSuite.scala

```diff
@@ -444,6 +448,14 @@ class JsonProtocolSuite extends SparkFunSuite {
         assertEquals(e1.executorInfo, e2.executorInfo)
       case (e1: SparkListenerExecutorRemoved, e2: SparkListenerExecutorRemoved) =>
         assert(e1.executorId == e1.executorId)
+      case (e1: SparkListenerExecutorMetricsUpdate, e2: SparkListenerExecutorMetricsUpdate) =>
+        assert(e1.execId == e2.execId)
+        assertSeqEquals[(Long, Int, Int, TaskMetrics)](e1.taskMetrics, e2.taskMetrics, (a, b) => {
+          assert(a._1==b._1)
+          assert(a._2==b._2)
+          assert(a._3==b._3)
```

Comment:
style: no space before and after `==`. Also please use `===` (3 equal signs) instead of `==` everywhere in this file.